File: mycollab-services-ondemand/src/main/java/com/mycollab/ondemand/configuration/DeploymentMode.java
package com.mycollab.ondemand.configuration;

import com.mycollab.configuration.IDeploymentMode;
import com.mycollab.configuration.ServerConfiguration;
import com.mycollab.configuration.SiteConfiguration;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.core.annotation.Order;
import org.springframework.stereotype.Service;

/**
 * @author MyCollab Ltd
 * @since 5.2.2
 */
@Order(value = 1)
@Service
public class DeploymentMode implements IDeploymentMode {

    @Autowired
    private ServerConfiguration serverConfiguration;

    @Override
    public boolean isDemandEdition() {
        return true;
    }

    @Override
    public boolean isCommunityEdition() {
        return false;
    }

    @Override
    public boolean isPremiumEdition() {
        return false;
    }

    @Override
    public String getSiteUrl(String subDomain) {
        return String.format(serverConfiguration.getAppUrl(), subDomain);
    }
}

File: mycollab-services-ondemand/src/main/java/com/mycollab/ondemand/configuration/DeploymentMode.kt
package com.mycollab.ondemand.configuration

import com.mycollab.configuration.IDeploymentMode
import com.mycollab.configuration.ServerConfiguration
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.InitializingBean
import org.springframework.core.annotation.Order
import org.springframework.stereotype.Component

/**
 * @author MyCollab Ltd
 * @since 5.2.2
 */
@Order(value = 1)
@Component
class DeploymentMode(private val serverConfiguration: ServerConfiguration) : IDeploymentMode, InitializingBean {
    override fun afterPropertiesSet() {
        LOG.info("Mode cloud $isDemandEdition")
    }

    override val isDemandEdition: Boolean
        get() = false

    override val isCommunityEdition: Boolean
        get() = false

    override val isPremiumEdition: Boolean
        get() = true

    override fun getSiteUrl(subDomain: String?) = String.format(serverConfiguration.siteUrl, subDomain)

    override fun getResourceDownloadUrl() = serverConfiguration.resourceDownloadUrl

    override fun getCdnUrl() = serverConfiguration.cdnUrl

    companion object {
        val LOG = LoggerFactory.getLogger(DeploymentMode::class.java)
    }
}
